runconfig/hostconfig_solaris.go

```diff
@@ -45,3 +45,17 @@ func ValidateIsolation(hc *container.HostConfig) error {
 func ValidateQoS(hc *container.HostConfig) error {
 	return nil
 }
+
+// ValidateCredentialSpec performs platform specific validation of
+// the credential spec in the hostconfig structure. Solaris does not support
+// this Windows-only feature.
+func ValidateCredentialSpec(hc *container.HostConfig) error {
+	// We may not be passed a host config, such as in the case of docker commit
+	if hc == nil {
```

Comment:
The best way to have it only once is to have a common `ValidateCredentialSpec` function in `hostconfig.go` utilising runtime.GOOS in the condition. I've had pushback about doing this having maintainers generally preferring _platform.go files, although sometimes it makes sense to inline it here. Happy to move to `hostconfig.go` if you think it's worth doing here. Haven't done that for the next update though.